### PR TITLE
Add `-DPYBIND11_WERROR=ON` to mingw cmake commands

### PR DESCRIPTION
**Pull request by rwgk**
## Description
Add `-DPYBIND11_WERROR=ON` to mingw cmake commands, plus a couple minor tweaks to resolve warnings.

Brings the two MINGW jobs on par with all other CI jobs.

## Suggested changelog entry:

```rst

```

### CODE REVIEW

**Comment by rwgk** [comment]
@henryiii this is ready for review now (I don't know if you get auto notifications, in addition to the auto-add as reviewer).

**Comment by rwgk** [comment]
@Skylion007
@henryiii 

Hoping you can take a look. This is a very simple PR, plugging an obvious hole in our `PYBIND11_WERROR` coverage.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -929,7 +929,7 @@ jobs:
       run: git clean -fdx
 
     - name: Configure C++14
-      run: cmake -G "MinGW Makefiles" -DCMAKE_CXX_STANDARD=14 -DDOWNLOAD_CATCH=ON -S . -B build2
+      run: cmake -G "MinGW Makefiles" -DCMAKE_CXX_STANDARD=14 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DPYBIND11_WERROR=ON -DDOWNLOAD_CATCH=ON -S . -B build2
```
Comment (henryiii) [inline .github/workflows/ci.yml:932]:
```suggestion
      run: cmake -G "MinGW Makefiles" -DCMAKE_CXX_STANDARD=14 -DCMAKE_VERBOSE_MAKEFILE=ON -DPYBIND11_WERROR=ON -DDOWNLOAD_CATCH=ON -S . -B build2
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -947,7 +947,7 @@ jobs:
       run: git clean -fdx
 
     - name: Configure C++17
-      run: cmake -G "MinGW Makefiles" -DCMAKE_CXX_STANDARD=17 -DDOWNLOAD_CATCH=ON -S . -B build3
+      run: cmake -G "MinGW Makefiles" -DCMAKE_CXX_STANDARD=17 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DPYBIND11_WERROR=ON -DDOWNLOAD_CATCH=ON -S . -B build3
```
Comment (henryiii) [inline .github/workflows/ci.yml:950]:
```suggestion
      run: cmake -G "MinGW Makefiles" -DCMAKE_CXX_STANDARD=17 -DCMAKE_VERBOSE_MAKEFILE=ON -DPYBIND11_WERROR=ON -DDOWNLOAD_CATCH=ON -S . -B build3
```

Let's be consistent, and this is really only to help cmake-gui anyway.

**Review by henryiii — APPROVED** [review]
Minor nit, but otherwise seems fine. Verbose makefile make logs a bit long, not sure I prefer them, but okay if you want to.

**Comment by rwgk** [comment]
Thanks for the review!

> Verbose makefile make logs a bit long, not sure I prefer them, but okay if you want to.

This setting is only for the CI. I often analyze the logs with scripts. Having the information in the logs makes the difference between day and night (seeing and being blind). Another way to look at this: reducing is easy and conclusive, guessing is hard and ambiguous.

**Pull request merged**